Link Format: collapse selection to end after insert

ellatrix opened the pull request:
## Description

Very small code change and change in behaviour.

With change:

<img width="104" alt="Screenshot 2019-08-21 at 14 58 58" src="https://user-images.githubusercontent.com/4710635/63433837-39da8600-c424-11e9-9baf-f77a80261aff.png">

## How has this been tested?

Insert a link.

## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->

ellatrix committed:
Link Format: collapse selection to end after insert

WunderBart reviewed (approved):
Tested locally, works as expected. Nice UX improvement, thanks! 🙌 


| **Before** | **After** |
|-----------|----------|
|![link bef](https://user-images.githubusercontent.com/1451471/76105354-1993a280-5fd5-11ea-8e3f-e3d4c673c809.gif)|![link aft](https://user-images.githubusercontent.com/1451471/76105367-21ebdd80-5fd5-11ea-8a67-161679574b0d.gif)|

mtias commented:
Let's get it in.

mkaz reviewed (approved):
👍 Looks good. I rebased it against master.

mkaz committed:
Fix rebase & e2e tests

The inserting of links cahnges, so need to
remove the click to apply, and add tab + enter.
